covid_graphs/covid_graphs/scatter_plot.py

```diff
@@ -21,50 +23,44 @@ def __str__(self):
-                    self.best_param = result.gamma2
-
-                self.best_error = result.summary.error
-
-            print(f"b_0={self.best_b0}, {self.param_name}={self.best_param} is the best fit "
-                  f"for prefix_length={self.prefix_length}, error={self.best_error}")
-
-        with open(country_proto, "rb") as f:
-            country_data = CountryData()
-            text_format.Parse(f.read(), country_data)
-            self.real_positive = [0] * self.prefix_length + [
-                day.positive for day in country_data.stats
-            ]
-            first_date = country_data.stats[0].date
-            first_date = datetime(day=first_date.day, month=first_date.month,
-                                  year=first_date.year) - timedelta(days=self.prefix_length)
-
-            self.date_list = [(first_date + timedelta(days=d)).strftime('%Y-%m-%d')
-                              for d in range(len(self.real_positive) + EXTENSION)]
+    def __init__(self, country_data_file: Path, simulation_pb2_file: Path):
+        simulation_results = SimulationResults()
+        simulation_results.ParseFromString(simulation_pb2_file.read_bytes())
+
+        self.best_error, self.best_b0, self.best_gamma2 = 1e20, None, None
+        for result in simulation_results.results:
+            if result.summary.error > self.best_error:
+                continue
+            runs = result.runs
+            self.simulated_positive = [run.daily_positive for run in runs]
+            self.positive_days = chain.from_iterable(
+                range(len(run.daily_positive)) for run in runs)
+            self.daily_infected = [run.daily_infected for run in runs]
+            self.infected_days = chain.from_iterable(
+                range(len(run.daily_infected)) for run in runs)
+
+            self.deltas = result.deltas
+            self.best_b0 = result.b0
+            self.prefix_length = result.prefix_length
+            if result.HasField("alpha"):
+                self.param_name = "alpha"
+                self.best_param = result.alpha
+            else:
+                self.param_name = "gamma_2"
+                self.best_param = result.gamma2
+
+            self.best_error = result.summary.error
+
+        print(f"b_0={self.best_b0}, {self.param_name}={self.best_param} is the best fit "
+              f"for prefix_length={self.prefix_length}, error={self.best_error}")
+
+        country_report = CountryReport(country_data_file)
+        # TODO: Complete Slovak data, so that we don't have to do this dance
+        self.daily_positive = np.concatenate((np.zeros(self.prefix_length),
+                                             country_report.daily_positive), 0)
+        first_date = country_report.dates[0] - timedelta(days=self.prefix_length)
+        self.date_list = [(first_date + timedelta(days=d)).strftime('%Y-%m-%d')
+                          for d in range(len(self.daily_positive) + EXTENSION)]
```

Comment:
I'm not sure what we are trying to achieve on these lines.

Reply:
It's related to the TODO: `# TODO: Complete Slovak data, so that we don't have to do this dance`

Our Slovak data starts late, because the number of tests is not part of the JHU database and we have to put it in by hand. I'll get to this eventually, but it's really low in priority.